app/models/auth_token.rb

```diff
@@ -84,34 +106,11 @@ def headers(_context)
 
   def self.find_token(api_version, retries: 0)
     auth_token = authorized.order(Arel.sql("RANDOM()")).first
-    return auth_token if auth_token.high_rate_limit?(api_version)
+    return auth_token if auth_token.safe_to_use?(api_version)
```

Comment:
If it's not safe to use, should we just stop instead of retrying a few more times?

Reply:
If we don't pass this check, it'll fetch a new random record next time around